pkg/runctl/controller.go

```diff
@@ -176,7 +176,17 @@ func (c *Controller) syncHandler(key string) error {
 		return nil
 	}
 
-	pipelineRun := k8s.NewPipelineRun(pipelineRunAPIObj, c.factory)
+	pipelineRun, err := k8s.NewPipelineRun(pipelineRunAPIObj, c.factory)
+
+	// If pipelineRun is not found there is nothing to sync
+	if pipelineRun == nil {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
```

Comment:
You should check the error first, because `pipelineRun` will be nil in case of an error.